CMakeLists.txt

```diff
@@ -199,41 +58,57 @@ elseif (UNIX)
   # link against the Python library. The resulting shared library will have
   # missing symbols, but that's perfectly fine -- they will be resolved at
   # import time.
-
-  # .SO file extension on Linux/Mac OS
-  set_target_properties(example PROPERTIES SUFFIX ".so")
-
-  # Optimize for a small binary size
-  if (NOT ${U_CMAKE_BUILD_TYPE} MATCHES DEBUG)
-    set_target_properties(example PROPERTIES COMPILE_FLAGS "-Os")
+  if(MSVC)
+    target_link_libraries(${target_name} ${PYTHON_LIBRARIES})
+  elseif(APPLE)
+    # Make sure OS X does not have any issues with missing symbols
+    set_target_properties(${target_name} PROPERTIES MACOSX_RPATH ".")
```

Comment:
The line existed previously, I just moved it into the function. It seems to date all the way back to the [initial commit](https://github.com/pybind/pybind11/commit/38bd71134a428d641d8ca34b3ea94358a387703d#diff-af3b638bc2a3e6c650974192a53c7291R97).
I had assumed it was related to the "-undefined dynamic_lookup" flag which is right after it. But now that I took a closer look, I don't think it's needed at all. What's more puzzling is that the CMake docs list this as an [ON/OFF property](https://cmake.org/cmake/help/v3.0/prop_tgt/MACOSX_RPATH.html), not a string.

I think it should be safe to remove it. I've never used it when compiling python modules with cmake. On the other hand, the "-undefined dynamic_lookup" flag definitely is required.